frontend/inst_pointer_chase_gen_test.py

```diff
@@ -0,0 +1,115 @@
+"""Tests for inst_pointer_chase_gen."""
+# Access to protected class members is common for unit tests.
+# pylint: disable=protected-access
+
+import unittest
+
+import cfg_pb2
+import inst_pointer_chase_gen
+
+
+def _PopNextFunction(function_list):
+    """Pops and returns the next function in the list."""
+    if not function_list:
+        raise IndexError("in _PopNextFunction: list of functions is empty")
+    return function_list.pop(0)
+
+
+class InstPointerChaseGeneratorTest(unittest.TestCase):
+
+    def setUp(self):
+        self.depth = 3
+        self.num_callchains = 2
+        self.gen = inst_pointer_chase_gen.InstPointerChaseGenerator(
+            self.depth, self.num_callchains, function_selector=_PopNextFunction)
+
+    def test_generate_callchain_mappings(self):
+        self.gen._GenerateCallchainMappings()
+        self.assertEqual(self.gen._caller2callee[0], 1)
+        self.assertEqual(self.gen._caller2callee[1], 2)
+        self.assertEqual(self.gen._caller2callee[2],
+                         inst_pointer_chase_gen.NO_CALLEE)
+        self.assertEqual(self.gen._caller2callee[3], 4)
+        self.assertEqual(self.gen._caller2callee[4], 5)
+        self.assertEqual(self.gen._caller2callee[5],
+                         inst_pointer_chase_gen.NO_CALLEE)
+
+    def test_generate_callchain_function_codeblockbodies(self):
+        self.gen._GenerateCallchainMappings()
+        self.gen._GenerateCallchainFunctions()
+        self.assertEqual(len(self.gen._functions),
+                         self.depth * self.num_callchains)
+
+        # All functions should use the same main code block body.
+        code_block_bodies = set()
+        for func_id, func in self.gen._functions.items():
+            code_block_bodies.add(func.instructions[0].code_block_body_id)
```

Comment:
Split this into a separate loop before line 53. Performance drop is minimal but it is easier to read as it is a separate concept.

Reply:
Done.